ERROR - 2022-12-04 03:11:25 --> Error connecting to the database: mysqli_sql_exception: No connection could be made because the target machine actively refused it.
 in E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\MySQLi\Connection.php:171
Stack trace:
#0 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\MySQLi\Connection.php(171): mysqli->real_connect('localhost', 'root', '', 'ci-app', 3306, '', 0)
#1 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\BaseConnection.php(382): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\BaseConnection.php(561): CodeIgniter\Database\BaseConnection->initialize()
#3 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\BaseBuilder.php(1585): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#4 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Model.php(223): CodeIgniter\Database\BaseBuilder->get()
#5 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\BaseModel.php(585): CodeIgniter\Model->doFindAll(0, 0)
#6 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\app\Controllers\UserCrud.php(15): CodeIgniter\BaseModel->findAll()
#7 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\CodeIgniter.php(927): App\Controllers\UserCrud->index()
#8 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\UserCrud))
#9 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#10 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\public\index.php(67): CodeIgniter\CodeIgniter->run()
#11 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Commands\Server\rewrite.php(46): require_once('E:\\xampp\\htdocs...')
#12 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: No connection could be made because the target machine actively refused it.
 in E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\MySQLi\Connection.php:215
Stack trace:
#0 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\BaseConnection.php(382): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\BaseConnection.php(561): CodeIgniter\Database\BaseConnection->initialize()
#2 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Database\BaseBuilder.php(1585): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Model.php(223): CodeIgniter\Database\BaseBuilder->get()
#4 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\BaseModel.php(585): CodeIgniter\Model->doFindAll(0, 0)
#5 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\app\Controllers\UserCrud.php(15): CodeIgniter\BaseModel->findAll()
#6 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\CodeIgniter.php(927): App\Controllers\UserCrud->index()
#7 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\UserCrud))
#8 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\public\index.php(67): CodeIgniter\CodeIgniter->run()
#10 E:\xampp\htdocs\CodeIgniter_Resource\ci-app\system\Commands\Server\rewrite.php(46): require_once('E:\\xampp\\htdocs...')
#11 {main}
CRITICAL - 2022-12-04 03:11:25 --> Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it.

in SYSTEMPATH\Database\BaseConnection.php on line 418.
 1 SYSTEMPATH\Database\BaseConnection.php(561): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1585): CodeIgniter\Database\BaseConnection->query('SELECT *
FROM `users`
ORDER BY `id` DESC', [], false)
 3 SYSTEMPATH\Model.php(223): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\BaseModel.php(585): CodeIgniter\Model->doFindAll(0, 0)
 5 APPPATH\Controllers\UserCrud.php(15): CodeIgniter\BaseModel->findAll()
 6 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\UserCrud->index()
 7 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\UserCrud))
 8 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2022-12-04 03:28:02 --> syntax error, unexpected '}', expecting ';'
in APPPATH\Controllers\UserCrud.php on line 22.
 1 SYSTEMPATH\Autoloader\Autoloader.php(259): CodeIgniter\Autoloader\Autoloader->includeFile('APPPATH\\Controllers\\UserCrud.php')
 2 SYSTEMPATH\Autoloader\Autoloader.php(237): CodeIgniter\Autoloader\Autoloader->loadInNamespace('App\\Controllers\\UserCrud')
 3 [internal function]: CodeIgniter\Autoloader\Autoloader->loadClass('App\\Controllers\\UserCrud')
 4 [internal function]: spl_autoload_call('App\\Controllers\\UserCrud')
 5 SYSTEMPATH\CodeIgniter.php(877): class_exists('\\App\\Controllers\\UserCrud', true)
 6 SYSTEMPATH\CodeIgniter.php(469): CodeIgniter\CodeIgniter->startController()
 7 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 8 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 9 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2022-12-04 03:28:18 --> syntax error, unexpected '}', expecting ';'
in APPPATH\Controllers\UserCrud.php on line 22.
 1 SYSTEMPATH\Autoloader\Autoloader.php(259): CodeIgniter\Autoloader\Autoloader->includeFile('APPPATH\\Controllers\\UserCrud.php')
 2 SYSTEMPATH\Autoloader\Autoloader.php(237): CodeIgniter\Autoloader\Autoloader->loadInNamespace('App\\Controllers\\UserCrud')
 3 [internal function]: CodeIgniter\Autoloader\Autoloader->loadClass('App\\Controllers\\UserCrud')
 4 [internal function]: spl_autoload_call('App\\Controllers\\UserCrud')
 5 SYSTEMPATH\CodeIgniter.php(877): class_exists('\\App\\Controllers\\UserCrud', true)
 6 SYSTEMPATH\CodeIgniter.php(469): CodeIgniter\CodeIgniter->startController()
 7 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 8 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 9 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2022-12-04 05:44:42 --> Invalid file: edit_user.php
in SYSTEMPATH\Exceptions\FrameworkException.php on line 33.
 1 SYSTEMPATH\View\View.php(201): CodeIgniter\Exceptions\FrameworkException::forInvalidFile('edit_user.php')
 2 SYSTEMPATH\Common.php(1134): CodeIgniter\View\View->render('edit_user', [], true)
 3 APPPATH\Controllers\UserCrud.php(46): view('edit_user', [...])
 4 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\UserCrud->edit('6')
 5 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\UserCrud))
 6 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 7 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 8 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
